### DISCUSSION TITLE
Where do I set the time lock  in  the taproot assets-level vUTXO script?

### DISCUSSION THREAD

**Discussion by fanshome1001** [lightninglabs/taproot-assets#901] [Q&A]
Where do I set the time lock  in  the taproot assets-level vUTXO script ("lock_time" and  "relative_lock_time")? My understanding should be when generating a new address ($ tapcli addrs new --asset_id ), but there is no parameter that can be set .

**Comment by dstadulis** [comment]
Will update you shortly regarding this

**Reply by shuishihang** [reply]
Has this issue been resolved? For Taproot Assets' time-lock mechanism, should it be implemented through swap transactions or by creating a script address? If it's possible to implement time-lock script address locking, could you tell me how to achieve it?

**Reply by guggero** [reply]
When you create the asset address (the `script_key`) to send assets to in order to lock them, you basically include a bitcoin script that uses the `OP_CHECKLOCKTIMEVERIFY` (for an absolute time lock) or `OP_CHECKSEQUENCEVERIFY` (for a relative time lock) codes in one of the tapscript leaves (or just a single leaf).

Then when you _spend_ the time locked funds again, you need to make sure you set the `LockTime` and/or `RelativeLockTime` fields correctly in the `vPacket.Output` correctly before signing them.

A full example can be found here: https://github.com/lightninglabs/taproot-assets/blob/53693935a0aef058531395f3ca5f9edb67d8be79/itest/psbt_test.go#L2589

**Comment by penglipeng666** [comment]
I do too
